crates/compiler/gen_wasm/src/low_level.rs

```diff
@@ -1625,7 +1625,10 @@ impl<'a> LowLevelCall<'a> {
                     .load_symbols(&mut backend.code_builder, &[num, bits]);
                 match CodeGenNumType::from(self.ret_layout) {
                     I32 => backend.code_builder.i32_shl(),
-                    I64 => backend.code_builder.i64_shl(),
+                    I64 => {
+                        backend.code_builder.i64_extend_u_i32();
```

Comment:
how does this work?

Reply:
`i64_extend_u_i32` is a Wasm typecast instruction.
`i64_shl`, a shift left instruction, takes two i64s. Previously we were passing it an i64 and an i32, and, for some reason, everybody was fine with this. No errors from Wasmer, Wasm3 or `wasm-validate`, etc.!
The only thing that complained was my little interpreter! 😊 
Obviously I thought I was wrong at first but no, the spec says it should be two i64s.
https://webassembly.github.io/spec/core/appendix/index-instructions.html

Same goes for all the other i64 bitshifts.

Reply:
ah, just looked up the signature of `shiftLeftBy : Int a, U8 -> Int a` and then that cast makes sense!

Reply:
yes that's right, I should have mentioned that!